doc/declarative-state-allocation/rds.md

```diff
@@ -0,0 +1,146 @@
+
+
+Creating a new instance
+---------------
+
+To create a new instance we have to add a ***rds-instance*** trait with allocateDatabase property as follows:
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: rds-test-owner
+  namespace: default
+spec:
+  components:
+  - name: rds-test-owner
+    properties:
+      cpu: "1"
+      exposeType: ClusterIP
+      image: nginx
+      memory: 1024Mi
+      ports:
+      - expose: true
+        port: 80
+        protocol: TCP
+    traits:
+    - type: scaler
+      properties:
+        replicas: 1
+    - type: rds-instance
+      properties:
+        # This trait definition will create a new RDS instance
+        allocateDatabase:
+          storage: 400
+          instanceClass: db.t3.medium
+        name: chainsync-qa-mainnet
+    type: webservice
+```
+Verifying resource:
+
+```shell
+# Please note that CrossPlane objects are not namespace scoped. This is not a bug!
+% kubectl get rdsinstance
+NAME                   READY   SYNCED   STATE       ENGINE     VERSION   AGE
+chainsync-qa-mainnet   True    True     available   postgres   15.4      11h
+# Ready and Synced = True, State is available
+# Everything is Okay!
+```
+
+Usually it takes up to 30 minutes to complete. Some relevant notes about the new instances:
+
+1. We are using a "Security by default" approach:<br>
+    1.1. Databases are not publicly available.<br>
+    1.2. In-transit encryption is supported.<br>
+    1.3. At-rest encryption is enabled.<br>
+    1.4. Passwords are randomly generated by AWS and saved in Kubernetes secrets.<br>
+    1.5. Automated backups are enabled.<br>
+2. Automatic minor version upgrades is disabled by default.
+3. Performance Insights is enabled by default.
+4. Multi-AZ is enabled by default.
+5. Default region is us-east-1 (North Virginia).
+
+This trait also has support for setting some optional properties:
+
+| Property  |  Description  |
+|---|---|
+| `backupRetentionPeriod` | Defines a backup retention period (Number of days)<br>|
+| `preferredBackupWindow` | Defines backup window schedule |
+| `preferredMaintenanceWindow` | Defines maintenance window schedule |
+| `iops` | Overrides IOPS settings |
+| `deletionProtection` | Enable deletion protection feature |
+| `customParameters` | Update RDS [parameter group](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.PostgreSQL.CommonDBATasks.Parameters.html)<br>*A list of all available parameters can be found [here](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.PostgreSQL.CommonDBATasks.Parameters.html#Appendix.PostgreSQL.CommonDBATasks.Parameters.parameters-list)* |
+
+**⚠️ IMPORTANT**!
+
+Without `deletionProtection`, the RDS instance is tied to the parent Application lifecycle so if the Application is deleted (or just recycled through VelaUX, for example), the RDS instance is deleted as well.
```

Comment:
Two thoughts:

1. I'm still not sure why we are exposing this option at all. What's the use case for declarative *allocation* of a db which outlives the declaration?
2. At a minimum, we should change the wording here, as it suggests that deletionProtection is something you should probably have if you're not sure, rather than the other way around